test/include/distributions/DistributionValuePostprocessor.h

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+#ifndef DISTRIBUTIONVALUEPOSTPROCESSOR_H
+#define DISTRIBUTIONVALUEPOSTPROCESSOR_H
+
+#include "GeneralPostprocessor.h"
+
+class DistributionValuePostprocessor;
+class Distribution;
+
+template<>
+InputParameters validParams<DistributionValuePostprocessor>();
+
+/**
+ * This postprocessor is used to test the distribution capabilities.
+ */
+class DistributionValuePostprocessor : public GeneralPostprocessor
+{
+public:
+  DistributionValuePostprocessor(const InputParameters & parameters);
+
+  virtual void initialize() override;
+  virtual void execute() override;
+  virtual PostprocessorValue getValue() override;
+
+protected:
+  /// Object of statistics distribution
+  Distribution & _distribution;
```

Comment:
This is where you are running into problems with restart tests. The Distribution seed will get reset if we restart. We could simply declare this as restartable data but I think a better solution would be to have MOOSE just automatically do this for all Distribution objects since this will be a universal problem.